sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/DefaultSynchronousMetricStorage.java

```diff
@@ -162,7 +178,11 @@ private AggregatorHandle<T, U> getAggregatorHandle(
               + metricDescriptor.getSourceInstrument().getName()
               + " has exceeded the maximum allowed cardinality ("
               + maxCardinality
-              + ").");
+              + ")."
+              + ((memoryMode == REUSABLE_DATA && aggregationTemporality == DELTA)
+                  ? "REUSABLE_DATA memory mode used, hence expiring unused Attributes might "
+                      + "result in increased memory allocations. See MemoryMode.java for more details."
+                  : ""));
```

Comment:
I don't think we should be logging details about our memory model to users.

Reply:
I can remove it, but they will have no easy way to find out why they suddenly have increased memory allocations (if they chose REUSABLE_DATA and DELTA temporality).

Reply:
Ok. I reverted the change in logger message